views-fieldset/src/main/java/io/micronaut/views/fields/render/views/FormElementRendererViewsConfiguration.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.views.fields.render.views;
+
+import io.micronaut.context.annotation.DefaultImplementation;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Nullable;
+import io.micronaut.views.fields.render.FormElementRenderer;
+
+/**
+ * Configuration for {@link FormElementRenderer} based on views.
+ * @author Sergio del Amo
+ * @since 4.1.0
+ */
+@Experimental
+@DefaultImplementation(FormElementRendererViewsConfigurationProperties.class)
+public interface FormElementRendererViewsConfiguration {
```

Comment:
could be simplified by records which can be a configuration properties nowadays

Reply:
I tried to use immutable configuration via https://github.com/micronaut-projects/micronaut-views/pull/647 but it fails with: 

```

io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.views.fields.render.views.FormElementRendererViewsConfiguration$Intercepted]

Message: At least one @Introduction method interceptor required, but missing for method: String getTrixEditor(). Check if your @Introduction stereotype annotation is marked with @Retention(RUNTIME) and @InterceptorBean(..) with the interceptor type. Otherwise do not load @Introduction beans if their interceptor definitions are missing!
Path Taken: new FormElementRendererViewsConfiguration$Intercepted(BeanResolutionContext $beanResolutionContext,BeanContext $beanContext,Qualifier $qualifier,List $interceptors)
	at io.micronaut.context.DefaultBeanContext.resolveByBeanFactory(DefaultBeanContext.java:2322)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2277)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2289)
	at io.micronaut.context.DefaultBeanContext.createRegistration(DefaultBeanContext.java:3056)
	at io.micronaut.context.SingletonScope.getOrCreate(SingletonScope.java:81)
	at io.micronaut.context.DefaultBeanContext.findOrCreateSingletonBeanRegistration(DefaultBeanContext.java:2958)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2919)
	at io.micronaut.context.DefaultBeanContext.resolveBeanRegistration(DefaultBeanContext.java:2730)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1693)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:840)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:832)
	at io.micronaut.views.fields.thymeleaf.FormElementRend
```

Reply:
probably you are missing the `micronaut-context` dependency. From a performance perspective records are better than interfaces

Reply:
> probably you are missing the micronaut-context dependency. 

Thanks. Adding [context](https://github.com/micronaut-projects/micronaut-core/tree/4.2.x/context) allowed me to use immutable configuration with an interface.

[I have changed the `@ConfigurationProperties` to be record](https://github.com/micronaut-projects/micronaut-views/pull/613/commits/f68fb0f6182b39d93d127f614df8598f30ddfa9c)